src/notebook/notebookKernel.ts

```diff
@@ -30,182 +28,144 @@ export class JuliaKernel implements vscode.NotebookKernel {
+	    )
+	    await Promise.all(cells.map(this.executeCell.bind(this)))
+	}
+
+	public dispose() {
+	    console.log(this.document.fileName)
+	    this._localDisposables.forEach(d => d.dispose())
+	}
+
+	async executeCell(cell: vscode.NotebookCell): Promise<void> {
+	    const task = vscode.notebook.createNotebookCellExecutionTask(cell.notebook.uri, cell.index, this.id)
+	    task.clearOutput()
+	    await this.start()
+	    const startTime = Date.now()
+	    task.start({startTime})
+	    const executionOrder = ++this._current_request_id
+	    task.executionOrder = executionOrder
+	    this.executionRequests.set(executionOrder, { startTime, task })
+	    this._msgConnection.sendNotification(notifyTypeRunCell, { current_request_id: this._current_request_id, code: cell.document.getText() })
+	}
+
+	public async start() {
+	    if (!this._terminal) {
+	        const connectedPromise = new Subject()
+	        const serverListeningPromise = new Subject()
+
+	        const pn = generatePipeName(uuid(), 'vscjl-nbk')
+
+	        const server = net.createServer(socket => {
+	            this._msgConnection = createMessageConnection(
+	                new StreamMessageReader(socket),
+	                new StreamMessageWriter(socket)
+	            )
+
+	            this._msgConnection.onNotification(notifyTypeRunCellSucceeded, ({ request_id }) => {
+	                const runEndTime = Date.now()
+	                const { task, startTime } = this.executionRequests.get(request_id)
+	                task.end({ success: true, duration: runEndTime - startTime })
+	            })
+
+	            this._msgConnection.onNotification(notifyTypeRunCellFailed, ({ request_id, output }) => {
+	                const runEndTime = Date.now()
+	                const { task, startTime } = this.executionRequests.get(request_id)
+	                task.appendOutput([new vscode.NotebookCellOutput([new vscode.NotebookCellOutputItem('application/x.notebook.error-traceback', output)])])
+	                task.end({ success: false, duration: runEndTime - startTime })
+	            })
+
+	            this._msgConnection.onNotification(notifyTypeDisplay, ({ mimetype, current_request_id, data }) => {
+	                const executionRequest = this.executionRequests.get(current_request_id)
+
+	                if (executionRequest) {
+	                    executionRequest.task.replaceOutput([new vscode.NotebookCellOutput([new vscode.NotebookCellOutputItem(mimetype, data)])])
+	                }
+	            })
+
+	            this._msgConnection.onNotification(notifyTypeStreamoutput, ({ name, current_request_id, data }) => {
+	                if (name === 'stdout') {
+	                    const executionRequest = this.executionRequests.get(current_request_id)
+	                    if (executionRequest) {
+	                        executionRequest.task.replaceOutput([new vscode.NotebookCellOutput([new vscode.NotebookCellOutputItem('application/x.notebook.stdout', data)])])
```

Comment:
I can see one problem with the append approach.
Assume you have a progress bar, the kernel will send the output for each tick,
however we could end up with 100s if not 1000s or more output objects for a single line progress bar.

In Jupyter extension we do not append stdout, as it could be a progress indicator.
We compress the ansi output by handling special characters used by terminals to generate progress bars.

I'm assuming we'd want the same fix here.
FYI - If we don't do that, then users could end up with a very large ipynb file with lots of unnecessary output.